modules/drivers/mongo/test/metabase/test/data/mongo.clj

```diff
@@ -36,11 +36,15 @@
     (ssl-required?) (merge (ssl-params))))
 
 (defmethod tx/dbdef->connection-details :mongo
-  [_ _ dbdef]
-  (conn-details {:dbname (tx/escaped-database-name dbdef)
-                 :user   "metabase"
-                 :pass   "metasample123"
-                 :host   "localhost"}))
+  [_driver _db-or-server dbdef]
+  (conn-details (merge
+                 {:dbname (tx/escaped-database-name dbdef)
+                  :host   (tx/db-test-env-var :mongo :host "localhost")
+                  :post   (Integer/parseUnsignedInt (tx/db-test-env-var :mongo :post "27017"))}
+                 (when-let [user (tx/db-test-env-var :mongo :user)]
+                   {:user user})
+                 (when-let [password (tx/db-test-env-var :mongo :password)]
+                   {:pass password}))))
```

Comment:
Don't hardcode the username and password used for the CI images. It broke local dev flows that use the Docker image in https://github.com/metabase/dev-scripts/blob/master/run-mongo.sh. It would also break things for people who installed MongoDB natively, unless they manually created that user and password. 

You can now configure user and password as needed by setting `MB_MONGO_TEST_USER` and `MB_MONGO_TEST_PASSWORD` respectively. This is in line with hour env vars work for our other driver tests. I also made `MB_MONGO_TEST_HOST` and `MB_MONGO_TEST_PORT` options as well instead of hardcoding them to `localhost` and `27017` respectively. This is also in line with other drivers. We're not using these options yet, but in the future it can be useful if we want to run two different versions of MongoDB locally with different ports or something like that.